packages/ripple-binary-codec/tools/generateDefinitions.js

```diff
@@ -0,0 +1,360 @@
+  path.join(process.argv[2], '/include/xrpl/protocol/TER.h'),
+)
+const transactionsMacroFile = readFile(
+  path.join(
+    process.argv[2],
+    '/include/xrpl/protocol/detail/transactions.macro',
+  ),
+)
+
+const capitalizationExceptions = {
+  NFTOKEN: 'NFToken',
+  URITOKEN: 'URIToken',
+  URI: 'URI',
+  UNL: 'UNL',
+  XCHAIN: 'XChain',
+  DID: 'DID',
+  ID: 'ID',
+  AMM: 'AMM',
+}
+
+// Translate from rippled string format to what the binary codecs expect
+function translate(inp) {
+  try {
+    if (inp.match(/^UINT/m))
+      if (inp.match(/256/m) || inp.match(/160/m) || inp.match(/128/m))
+        return inp.replace('UINT', 'Hash')
+      else return inp.replace('UINT', 'UInt')
+    if (inp == 'OBJECT' || inp == 'ARRAY')
+      return 'ST' + inp.substr(0, 1).toUpperCase() + inp.substr(1).toLowerCase()
+    if (inp == 'AMM') return inp
+    if (inp == 'ACCOUNT') return 'AccountID'
+    if (inp == 'LEDGERENTRY') return 'LedgerEntry'
+    if (inp == 'NOTPRESENT') return 'NotPresent'
+    if (inp == 'PATHSET') return 'PathSet'
+    if (inp == 'VL') return 'Blob'
+    if (inp == 'DIR_NODE') return 'DirectoryNode'
+    if (inp == 'PAYCHAN') return 'PayChannel'
+
+    const parts = inp.split('_')
+    let result = ''
+    for (x in parts)
+      if (capitalizationExceptions[parts[x]] != null) {
+        result += capitalizationExceptions[parts[x]]
+      } else
+        result +=
+          parts[x].substr(0, 1).toUpperCase() + parts[x].substr(1).toLowerCase()
+    return result
+  } catch (e) {
+    console.error(e, 'inp="' + inp + '"')
+  }
+}
+
+let output = ''
+function addLine(line) {
+  output += line + '\n'
+}
+
+////////////////////////////////////////////////////////////////////////
+//  Serialized type processing
+////////////////////////////////////////////////////////////////////////
+addLine('{')
+addLine('  "TYPES": {')
+addLine('    "Done": -1,')
+
+let stypeHits = [
+  ...sfieldHeaderFile.matchAll(
+    /^ *STYPE\(STI_([^ ]*?) *, *([0-9-]+) *\) *\\?$/gm,
+  ),
+]
+if (stypeHits.length === 0)
+  stypeHits = [
+    ...sfieldHeaderFile.matchAll(/^ *STI_([^ ]*?) *= *([0-9-]+) *,?$/gm),
+  ]
+for (let x = 0; x < stypeHits.length; ++x) {
+  addLine(
+    '    "' +
+      translate(stypeHits[x][1]) +
+      '": ' +
+      stypeHits[x][2] +
+      (x < stypeHits.length - 1 ? ',' : ''),
+  )
+}
+
+addLine('  },')
+
+////////////////////////////////////////////////////////////////////////
+//  Ledger entry type processing
+////////////////////////////////////////////////////////////////////////
+addLine('  "LEDGER_ENTRY_TYPES": {')
+addLine('    "Any": -3,')
+addLine('    "Child": -2,')
+addLine('    "Invalid": -1,')
+
+const unhex = (x) => {
+  x = ('' + x).trim()
+  if (x.substr(0, 2) == '0x') return '' + parseInt(x)
+  if (x.substr(0, 1) == "'" && x.length == 3) return x.charCodeAt(1)
+  return x
+}
+hits = [
+  ...ledgerFormatsMacroFile.matchAll(
+    /^ *LEDGER_ENTRY\(lt[A-Z_]+ *, *([x0-9a-f]+) *, *([^,]+), \({$/gm,
```

Comment:
there is no need for the `_` character in the first match group. In the rippled source code, I haven't found any instances of `LEDGER_ENTRY(lt_...` ledger-entry-types.

Reply:
There are some that have a `_` in the middle, which this match group includes - such as `ltSIGNER_LIST`.

Reply:
ah nice, thanks